packages/e2e-test-utils-playwright/src/page-utils/drag-files.ts

```diff
@@ -99,14 +100,19 @@ async function dragFiles(
 		/**
 		 * Drag the files over an element (fires `dragenter` and `dragover` events).
 		 *
-		 * @param  selector         A selector to search for an element.
-		 * @param  options          The optional options.
-		 * @param  options.position A point to use relative to the top-left corner of element padding box. If not specified, uses some visible point of the element.
+		 * @param  selectorOrLocator A selector or a locator to search for an element.
+		 * @param  options           The optional options.
+		 * @param  options.position  A point to use relative to the top-left corner of element padding box. If not specified, uses some visible point of the element.
 		 */
-		dragOver: async ( selector: string, options: Options = {} ) => {
-			const boundingBox = await this.page
-				.locator( selector )
-				.boundingBox();
+		dragOver: async (
```

Comment:
Why are we using Playwright here instead of Puppeteer if in Puppeteer drag and drop works out of the box?

Reply:
Several reasons:

1. We're in the middle of the migration from Puppeteer to Playwright. In order to make the migration path smoother, it makes sense for me to add some missing features in `e2e-test-utils-playwright`. Once these utils are implemented, I believe it will be easier for contributors to pick up or add more tests.
2. Puppeteer doesn't fully support drag-and-drop out of the box either. Specifically, if I'm not mistaken, Puppeteer also doesn't support API for dragging files (https://github.com/puppeteer/puppeteer/issues/7330). It seems like changes need to be made in Chrome Devtools Protocol to support this. If we implement it in Puppeteer, we would still have to implement these utils anyway.

@talldan has opened an issue (https://github.com/microsoft/playwright/issues/16437) in the Playwright repo already, I believe that's a good start. We can iterate and see where it goes from there ;)